piksi_tools/console/settings_view.py

```diff
@@ -222,33 +182,43 @@ def revert_to_prior_value(self, name, old, new, error_value=SBP_WRITE_STATUS.TIM
+            self.confirmed_set = False
+            res = self.settings.s_api.write(self.section, self.name, new)
+            if res == settings_write_res_e.SETTINGS_WR_OK:
+                self.value = new
+            else:
+                self.revert_to_prior_value(self.name, old, new, res)
+
+            self.confirmed_set = True
+
+        self.write_event.set()
+
+        # If we have toggled the Inertial Nav enable setting (currently "output mode")
+        # we display some helpful hints for the user
+        if (self.section == "ins" and self.name == "output_mode" and
+                old is not None and self.settings is not None):
+            if new in ['GNSS and INS', 'INS Only', 'Loosely Coupled', 'LC + GNSS', 'Debug', 'debug']:
+                hint_thread = threading.Thread(
+                    target=self.settings._display_ins_settings_hint)
+                hint_thread.start()
+            # regardless of which way setting is going, a restart is required
+            else:
+                self.settings.display_ins_output_hint()
 
     def _value_changed(self, name, old, new):
-        '''When a user changes a value, kick off a timed revert thread to revert it in GUI if no confirmation
-            that the change was successful is received.'''
-        if not self._prevent_revert_thread:
-            if getattr(self, 'settings', None):
-                if (old != new and old is not Undefined and new is not Undefined):
-                    if type(self.value) == unicode:
-                        self.value = self.value.encode('ascii', 'replace')
-                    self.confirmed_set = False
-                    self.timed_revert_thread = TimedDelayStoppableThread(
-                        SETTINGS_REVERT_TIMEOUT,
-                        target=self.revert_to_prior_value,
-                        args=(name, old, new))
-                    self.settings.set(self.section, self.name, self.value)
-                    self.timed_revert_thread.start()
-                # If we have toggled the Inertial Nav enable setting (currently "output mode")
-                # we display some helpful hints for the user
-                if (self.section == "ins" and self.name == "output_mode" and
-                        old is not None and self.settings is not None):
-                    if new in ['GNSS and INS', 'INS Only', 'Loosely Coupled', 'LC + GNSS', 'Debug', 'debug']:
-                        hint_thread = threading.Thread(
-                            target=self.settings._display_ins_settings_hint)
-                        hint_thread.start()
-                    # regardless of which way setting is going, a restart is required
-                    else:
-                        self.settings.display_ins_output_hint()
+        if getattr(self, 'settings', None) is None:
+            return
+
+        if self.reverting:
+            return
+
+        threading.Thread(target=self._write_value, args=(old, new)).start()
```

Comment:
I'm assuming most of the logic in`TimedDelayStoppableThread` went into libsettings, otherwise we're losing a lot of what looks like retry logic.  I would appreciate some verbiage in the final commit message for this PR to mention this.

Also assuming that libsettings probably doesn't deal with threads, so we put it into a thread to prevent it from blocking the UI?  Do we want to make a task queue for settings writes instead or is it OK to spawn one thread per settings write?

Reply:
Please make sure to test this thoroughly when a device is navigating with 5 Hz observations and default settings on a real serial port.  You cannot imagine the amount of testing I did on a real device when I added in the retry and logic for making sure a settings write took.  I think it is also important to test with firmware 2.0 that had a very lossy SBP interface due to nanomsg implementation.  I will plan to test on my end first thing next week to see if I notice anything odd.

Reply:
I might still be on the vacation mode as I don't see the retry logic for writing? Only in the indexed read case which admittedly needs to be added to libsettings.

Misc comments:
- libsettings doesn't support multithreading currently
- Thread is created to avoid blocking UI

Reply:
Write retry logic is in libsettings, see comment above: https://github.com/swift-nav/piksi_tools/pull/938/files#r244578599